Challenge/View/Movies/MoviesList/MoviesListView.swift

```diff
@@ -0,0 +1,259 @@
+//  Challenge
+//
+//  Created by Lucas Henrique Machado da Silva on 06/05/2018.
+//  Copyright © 2018 Lucas Henrique Machado da Silva. All rights reserved.
+//
+
+import UIKit
+
+class MoviesListView: UIViewController {
+    // MARK: - Outlets
+    @IBOutlet weak var tvMovies: UITableView!
+    @IBOutlet weak var aivLoader: UIActivityIndicatorView!
+    @IBOutlet weak var lMessage: UILabel!
+    @IBOutlet weak var lMore: UILabel!
+    @IBOutlet weak var sbName: UISearchBar!
+    
+    // MARK: - Constants
+    private let refreshControl = UIRefreshControl()
+    
+    // MARK: - Presenter
+    var presenter: MoviesListPresenter?
+    
+    // MARK: - Variables
+    var movies: [Movie]?
+    var selectedMovie: Movie?
+    var loadingMoreMovies: Bool = false
+    var allPagesLoaded: Bool = false
+    
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.presenter = MoviesListPresenter(self)
+        
+        self.setupView()
+    }
+    
+    // MARK: - Setups
+    func setupView() {
+        self.aivLoader.startAnimating()
+        
+        self.tvMovies.accessibilityIdentifier = "tvMovies"
+        self.tvMovies.allowsSelection = true
+        if #available(iOS 10.0, *) {
+            self.tvMovies.refreshControl = self.refreshControl
+        } else {
+            self.tvMovies.insertSubview(refreshControl, at: 0)
+        }
+        self.refreshControl.tintColor = UIColor(red: 145 / 255, green: 142 / 255, blue: 244 / 255, alpha: 1.0)
+        self.refreshControl.addTarget(self, action: #selector(self.refreshTableView), for: .valueChanged)
+        
+        self.presenter?.getUpcomingMovies()
+    }
+    
+    // MARK: - Navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if (segue.identifier == "ListToDetailSegue") {
+            guard let detailView = segue.destination as? MovieDetailView else {
+                return
+            }
+            
+            detailView.movie = selectedMovie
+            self.selectedMovie = nil
+        }
+    }
+    
+    // MARK: - Helpers
+    func showAlertView(with title: String, and message: String) {
+        let alertView = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        let alertAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+        alertView.addAction(alertAction)
+        
+        self.present(alertView, animated: true, completion: nil)
+    }
+    
+    @objc private func refreshTableView() {
+        self.presenter?.reloadAllMovies()
+    }
+    
+    func loadMoreMovies() {
+        self.aivLoader.startAnimating()
+        self.lMore.isHidden = false
+        self.presenter?.getUpcomingMovies()
+    }
+    
+    func displayAllPagesLoaded() {
+        self.lMore.text = "no more upcoming movies to load"
+        self.lMore.isHidden = false
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```

Comment:
Why this delay is necessary here ?

Reply:
@arctouch-ruannhomem It's used so the user can see the message before I hide it (I don't want this message to be forever in the user's screen).